File: FRONTEND/src/app/models/event.ts
import { Category } from "./category";
import { User } from "./user";

export class Event {

  public id: number | undefined;
  public name: string;
  public description: string;
  public imageUrl: string;
  public places: number | undefined;
  public startDate : Date | null;
  public endDate : Date | null;
  public isAvailable: boolean;
  public isApproved: boolean;
  public organizer: User;
  public category: Category;


  constructor() {
    this.id = undefined;
    this.name = '';
    this.description = '';
    this.imageUrl = '';
    this.places = undefined;
    this.startDate = null;
    this.endDate = null;
    this.isAvailable = false;
    this.isApproved = false;
    this.organizer = new User();
    this.category = new Category();

  }

}

File: BACKEND/src/main/java/com/application/entities/Event.java
package com.application.entities;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Setter
@Getter
@Table(name = "events")
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    private String description;
    private String imageUrl;
    private Integer places;
    private Date startDate;
    private Date endDate;
    private Boolean isAvailable;
    private Boolean isApproved;
    @ManyToOne
    private Category category;
    @ManyToOne
    private Organizer organizer;
    @OneToMany(mappedBy = "event", cascade = CascadeType.REMOVE)
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private List<Reservation> reservations = new ArrayList<>();

    public Event() { }

}
